src/watch.rs

```diff
@@ -51,14 +53,41 @@ impl WatchPathBuf {
 /// react to changes when they occur.
 pub struct Watch {
     /// Event receiver. Process using `Watch::process`.
-    pub rx: chan::Receiver<notify::Result<notify::Event>>,
-    /// OS-based notification when any file we watched changes.
-    notify: RecommendedWatcher,
-    /// The list of files we are watching;
+    pub event_rx: chan::Receiver<Vec<PathBuf>>,
+    watch_tx: chan::Sender<Vec<WatchPathBuf>>,
+}
+
+impl Watch {
+    /// Instantiate a new Watch.
+    pub fn new(logger: slog::Logger) -> Result<Watch, notify::Error> {
+        let (ntfy_tx, notify_rx) = chan::unbounded();
+        let (event_tx, event_rx) = chan::unbounded();
+        let (watch_tx, watch_rx) = chan::unbounded();
+
+        let filter = Filter {
+            notify: Watcher::new(ntfy_tx, Config::default())?,
```

Comment:
Is there a reason why the new watcher uses a `30s` poll interval, as opposed to the `0.1s` from before?

Reply:
I think we can probably increase it up to 3 seconds to take some pressure off of bigger projects, but 30s would kind of destroy the idea imo.

Reply:
The duration isn't important. I don't have a problem doing something other than the default config, but:

* We aren't using Notify's debounce anymore, so that delay isn't in the config
* I think that everywhere we care about, Notify is using OS level FS events, not polling. Test behavior bears that out.

Reply:
yeah, true, I figured out it’s only relevant for the poll backend.